packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx

```diff
@@ -42,11 +44,24 @@ const NotificationPaymentRecipient: React.FC<Props> = ({
   landingSiteUrl,
   getPaymentAttachmentAction,
   onPayClick,
-  handleReloadPayment,
+  handleFetchPaymentsInfo, // Maybe rename this props to fetchPayments or something like that
 }) => {
   const { pagoPaF24, f24Only } = payments;
+  const [paginationData, setPaginationData] = useState<PaginationData>({
+    page: 0,
+    size: 5,
+    totalElements: payments.pagoPaF24.length,
+  });
 
-  const isSinglePayment = pagoPaF24.length === 1 && !isCancelled;
+  const paginatedPayments = pagoPaF24.slice(
+    paginationData.page * paginationData.size,
+    (paginationData.page + 1) * paginationData.size
+  );
+
+  const isSinglePayment =
```

Comment:
I'ld split the singlePayment condition from requiredCondition.
We can have more than one payments and no required payment.
So the required condition will be the result of the some() method